app/util/concurrency.py

```diff
@@ -0,0 +1,64 @@
+from threading import Thread
+from datetime import datetime
+
+DEFAULT_TIMEOUT = 10
+
+"""
+Class which extends Thread to expose a result from the target (input) function.
+"""
+class ResultThread(Thread):
+    def run(self):
+        self.result = self._target()
```

Comment:
Generally you're not supposed to call `_`-prefixed properties directly (since Python doesn't have private methods, it's just done by convention). I don't know anything about python threads, but this seems like a red flag.

Reply:
Reading through this more, is the `target` function not supposed to have a return value? We could also like construct a bunch of functions that store their values in an array (although that sounds like more overhead than we want...).